aten/src/ATen/native/transformers/cuda/attention.cu

```diff
@@ -841,12 +845,15 @@ std::tuple<Tensor, Tensor> _flash_attention_forward(
       softmax_scale,
       false, /*zero_tensors = false for all calls here*/
       is_causal,
-      num_splits,
-      c10::nullopt);
-  return std::make_tuple(output, logsumexp);
+      return_debug_mask, /*return_softmax (this is used for testing)*/
+      num_splits);
+
+  debug_attn_mask = return_debug_mask ? debug_attn_mask : at::empty({0}, query.options());
+  // TODO: Need to check if this cast is okay
+  return std::make_tuple(output, logsumexp, (int64_t)philox_seed, (int64_t)philox_offset, debug_attn_mask);
```

Comment:
First cast is here. This casts the return type from uint64_t to int64_t. Native_functions.yaml doesn't currently have support for unit64_t afaik.

Reply:
This isn't safe. If the uint64_t is larger than int64_t I this will return a strange value. We want to store the actual bits verbatim in a different type, not do type conversion.